### ISSUE TITLE
Feature request: YaCy plugin/integration

### ISSUE THREAD

**Issue by SEVENID** [boramalper/magnetico#46]
YaCy ([site](http://yacy.net), [wiki](https://en.wikipedia.org/wiki/YaCy), [github](https://github.com/yacy)) is a free distributed search engine, built on principles of peer-to-peer (P2P) networks.
Advantages: Customizable good search interface exists;
Global indexed, not just a local database to speed up and increase the search area (syncing and crawling schedulers are included); 
Mutual increase in the popularity of projects (yacy users can search in magnetico, magnetico users - in www without additional settings).
Disadvantages: Not so much information about internal architecture, fast googling shows nothing. E.g.: Unknown (for me) yacy crawler plugin interface, I'm not sure it exists.

**Comment by boramalper** [comment]
Thanks for the suggestion, it sounds really interesting! Do you know how to make this integration happen? I mean, any interface / protocol to implement?

Also,

> magnetico users [can search] - in www without additional settings).

I think that would be far too away from the objective of the project, don't you think so? =)

**Comment by SEVENID** [comment]
> Do you know how to make this integration happen? I mean, any interface / protocol to implement?

Unfortunately I don't know, but [yacy grid looks like that.](https://github.com/yacy/yacy_grid_mcp) As I understood, magnetico is possible to connect to it a database (yacy has its own) and/or crawler as microservices.

> far too away from the objective of the project

Maybe. But yacy can find a convenient description in www for the link, link by main tags and not just a list of file names (that may not correspond to the content).

**Comment by heresjonny1** [comment]
I agree, YaCy is pretty far outside the scope of the project, since both projects have considerably different goals (as well as programming languages)

However, you might want to consider the creation of a Searx plugin. Searx can also pull from yacy, so it's an all-in-one solution. This would be a feature request for Searx, not this project.

https://asciimoo.github.io/searx/
